lib/digital_public_works_web/controllers/project_invite_controller.ex

```diff
@@ -0,0 +1,153 @@
+defmodule DigitalPublicWorksWeb.ProjectInviteController do
+  use DigitalPublicWorksWeb, :controller
+
+  alias DigitalPublicWorks.{Projects, Accounts}
+  alias DigitalPublicWorks.Repo
+  alias DigitalPublicWorksWeb.{Email, Mailer}
+
+  plug :get_project
+  plug :get_project_invite
+  plug :check_auth
+
+  defp get_project(%{params: %{"project_id" => id}} = conn, _args) do
+    conn |> assign(:project, Projects.get_project!(id))
+  end
+
+  defp get_project(conn, _args), do: conn
+
+  defp get_project_invite(%{params: %{"id" => id}} = conn, _args) do
+    try do
+      project_invite = Accounts.get_project_invite!(id)
+
+      project_invite =
+        if project = conn.assigns[:project] do
+          Map.put(project_invite, :project, project)
+        else
+          project_invite |> Repo.preload(:project)
+        end
+
+      conn |> assign(:project_invite, project_invite)
+    rescue
+      _ ->
+        conn
+        |> put_flash(:error, "Invite not found.")
+        |> redirect(to: Routes.page_path(conn, :index))
+        |> halt()
+    end
+  end
+
+  defp get_project_invite(conn, _args) do
+    project = conn.assigns.project
+
+    project_invite = Ecto.build_assoc(project, :project_invites, project: project)
+
+    conn |> assign(:project_invite, project_invite)
+  end
+
+  defp check_auth(conn, _args) do
+    cond do
+      can?(conn.assigns.current_user, action_name(conn), conn.assigns.project_invite) ->
+        conn
+
+      conn.assigns.current_user ->
+        conn
+        |> put_flash(:error, "You don't have access to that")
+        |> redirect(to: Routes.page_path(conn, :index))
+        |> halt()
+
+      true ->
+        conn
+        |> put_flash(:info, "You need to log in first")
+        |> redirect(to: Routes.session_path(conn, :new))
+        |> halt()
+    end
+  end
+
+  def index(conn, _params) do
+    project = conn.assigns.project
+
+    conn
+    |> assign(:project_invites, project |> project_invites)
+    |> assign(:users, project |> project_users)
+    |> assign(:changeset, Accounts.change_project_invite())
+    |> render("index.html")
+  end
+
+  def create(conn, %{"project_invite" => params}) do
+    project = conn.assigns.project
+
+    case Accounts.create_project_invite(conn.assigns.project, params) do
+      {:ok, project_invite} ->
+        Email.project_invite_email(project_invite |> Map.put(:project, project)) |> Mailer.deliver_now
```

Comment:
Suggest using deliver_later here just in case sendgrid gets slow. Can ultimately attempt a retry strategy later, but slowing down the request for sendgrid doesn't seem super worth it. Could also add sentry.io to catch delivery failures in the meantime.

Reply:
I keep going back and forth on this one. Sendgrid will slow down the response for this request, but since the email delivery will still be proceed on the same dyno will this actually negatively impact performance? 

Since the built in deliver_later strategy doesn't have retry, and since the request will be slowed down but overall performance shouldn't suffer,  I thought it would make sense to show the user an error (in this case a 500 screen most likely) to at least give them an idea something went wrong.

Ultimately, I think the best solution is implementing a database backed deliver later strategy with auto retries, and displaying the message status to the users in the UI.